src/java.desktop/share/classes/javax/swing/text/html/HTMLEditorKit.java

```diff
@@ -684,9 +685,11 @@ public AccessibleContext getAccessibleContext() {
         if (theEditor == null) {
             return null;
         }
-        if (accessibleContext == null) {
+        if (accessibleContext == null
+            || doc != theEditor.getDocument()) {
             AccessibleHTML a = new AccessibleHTML(theEditor);
```

Comment:
The "AccessibleHTML" internally registers the listener on the editor to update the state/doc. Why that notification does not work and it is necessary to recreate the "AccessibleHTML" here?

Reply:
The "AccessibleHTML" does internally register the listener on the editor to update the state/doc.

But the `rootHTMLAccessibleContext` is not updated/created for the new root element on doc change and when  `getAccessibleChildrenCount` is called it returns the child count 0, because the API is get called on the old object.

Reply:
if the "rootHTMLAccessibleContext" depends on the doc then I think it should be updated in that listener at the same moment the doc is replaced. especially taking into account that the "rootElementInfo" is updated already.

Reply:
In that case I think the `rootElementInfo` needs to update for `rootHTMLAccessibleContext` everytime whenever doc change happened.

Reply:
@mrserb  The `elementInfo` variable is updated for `rootHTMLAccessibleContext` while setting up the `rootElementInfo` on doc change.
 It does report the children correctly and as of now there is no need to re-create the accessibleContext everytime on doc change.

Reply:
@mrserb Please review the latest changes.